src/classes/AuthenticationSession.ts

```diff
@@ -32,6 +42,59 @@ export class AuthenticationSession {
     this.promptForPassword();
   }
 
+  public obtainNewAuthTokenUsingRefreshToken(): void {
+    if (!AuthenticationSession.sftpFusionAuthClientId) {
+      logger.error(
+        'Cannot obtain new access token without sftp client id.',
+      );
+      return;
+    }
+
+    if (!AuthenticationSession.sftpFusionAuthClientSecret) {
+      logger.error(
+        'Cannot obtain new access token without sftp client secret.',
+      );
+      return;
+    }
+
+    this.fusionAuthClient.exchangeRefreshTokenForAccessToken(
+      this.refreshToken,
+      AuthenticationSession.sftpFusionAuthClientId,
+      AuthenticationSession.sftpFusionAuthClientSecret,
+      '',
+      '',
+    )
+      .then((clientResponse) => {
+        this.authToken = clientResponse.response.access_token ?? '';
+        // The exchange refresh token for access token endpoint dose not return a timestamp,
+        // it returns expires_in in seconds.
+        // So we need to create the timestamp to be consistent with what is first
+        // returned upon initial authentication
+        this.authTokenExpiresAt = Date.now() + (clientResponse
+          .response.expires_in ?? 1 * 1000);
+        logger.info('New access token obtained');
+      })
+      .catch((clientResponse: unknown) => {
+        const message = isPartialClientResponse(clientResponse)
+          ? clientResponse.exception.error_description
+          : '';
+        logger.warn(`Error obtaining refresh token : ${message}`);
+        this.authContext.reject();
+      });
+  }
+
+  public tokenExpired(): boolean {
+    const expirationDate = new Date(this.authTokenExpiresAt);
+    return expirationDate <= new Date();
+  }
+
+  public tokenWouldExpireSoon(minutes = 5): boolean {
```

Comment:
Some thoughts here:

(1) It seems like we could combine these two flavors of "stale" into a single function.  If a token has only 4 minutes left or a token expired (has negative minutes left) in either case it would be less than the `minutes` and as far as I can tell the difference between "stale" vs "expired" has no logical impact at this point.

(2) `minutes` is generic (which makes the check at the end a bit hard to parse) -- maybe something like `staleThreshold`

(3) It is odd to me for duration to be passed in terms of minutes.  I'd make this seconds or milliseconds, which is more intuitive in the land of JS.